### ISSUE TITLE
New Check: FieldMustBeConstantCheck - Logger must be declared as private static final

### ISSUE THREAD

**Issue by vorburger** [sevntu-checkstyle/sevntu.checkstyle#481]
continuing on from https://github.com/sevntu-checkstyle/sevntu.checkstyle/issues/457, assuming that we can conclude and merge https://github.com/sevntu-checkstyle/sevntu.checkstyle/issues/461 first, here's the next one, this is a generalization of "Logger must be declared as private static final":

_Describe its behaviour with examples of code, with violation:_

```
import org.slf4j.Logger;

class NOK {
    Logger logger; // <= Checkstyle violation raised here!
}
```

and without:

```
import org.slf4j.Logger;

class OK {
    private static final Logger logger;
}
```

_Examples on how configuration will look like:_

```
<module name="TreeWalker">
    <module name="com.github.sevntu.checkstyle.checks.coding.FieldMustBeConstantCheck">
        <property name="className" value="org.slf4j.Logger" />
        <property name="scope" value="private" />
    </module>
</module>
```

@romani please feel free to just go ahead and edit this "spec" as you see fit..

**Comment by vorburger** [comment]
@romani ping, how about this idea?

**Comment by romani** [comment]
idea is ok.
User just need to use this Check with caution , as this rule is correct in most cases, but there will be a lot false-positives and need to be suppressed.
In javadoc we need to give user guidance links when static is good - https://wiki.apache.org/commons/Logging/StaticLog

We have a rule to apply all Checks to our own code, this rule cannot be applied for our Logging - as we are library - static is not ok. We might find some other case of usage.

**Comment by vorburger** [comment]
@romani how about a boolean option for isStatic on this? BTW: For link to refer to in JavaDoc see also http://slf4j.org/faq.html#declared_static ...

PS: I'll look at working on this, or another one of the different Checks proposed in https://github.com/sevntu-checkstyle/sevntu.checkstyle/issues/457, once we have fully concluded and merged https://github.com/sevntu-checkstyle/sevntu.checkstyle/pull/480 ...

**Comment by romani** [comment]
> how about a boolean option for isStatic on this?

good idea.

> BTW: For link to refer to in JavaDoc see also

good. It is good to give user well stated analysis.
